src/main/kotlin/io/snyk/plugin/ui/toolwindow/panels/SnykCodeDataflowPanel.kt

```diff
@@ -0,0 +1,153 @@
+package io.snyk.plugin.ui.toolwindow.panels
+
+import com.intellij.openapi.project.Project
+import com.intellij.ui.HyperlinkLabel
+import com.intellij.uiDesigner.core.GridConstraints
+import com.intellij.uiDesigner.core.GridLayoutManager
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.UIUtil
+import io.snyk.plugin.getDocument
+import io.snyk.plugin.navigateToSource
+import io.snyk.plugin.toVirtualFile
+import io.snyk.plugin.ui.baseGridConstraintsAnchorWest
+import snyk.common.lsp.DataFlow
+import snyk.common.lsp.IssueData
+import java.awt.Font
+import javax.swing.JPanel
+import javax.swing.JTextArea
+import javax.swing.event.HyperlinkEvent
+import kotlin.math.max
+
+class SnykCodeDataflowPanel(private val project: Project, codeIssueData: IssueData): JPanel() {
+    init {
+        val dataFlow = codeIssueData.dataFlow
+
+        this.layout = GridLayoutManager(1 + dataFlow.size, 1, JBUI.emptyInsets(), -1, 5)
+
+        this.add(
+            defaultFontLabel("Data Flow - ${dataFlow.size} step${if (dataFlow.size > 1) "s" else ""}", true),
+            baseGridConstraintsAnchorWest(0)
+        )
+
+        this.add(
+            stepsPanel(dataFlow),
+            baseGridConstraintsAnchorWest(1)
+        )
+    }
+
+
+    private fun stepsPanel(dataflow: List<DataFlow>): JPanel {
+        val panel = JPanel()
+        panel.layout = GridLayoutManager(dataflow.size, 1, JBUI.emptyInsets(), 0, 0)
+        panel.background = UIUtil.getTextFieldBackground()
+
+        val maxFilenameLength = dataflow.asSequence()
+            .filter { it.filePath.isNotEmpty() }
+            .map { it.filePath.substringAfterLast('/', "").length }
```

Comment:
it.filePath.substringAfterLast('/', "").length 
Is this safe for IntelliJ users on Windows machines? 


On Windows, file paths backslash \ is used as the directory separator instead of the forward slash.

Reply:
I think you might be right, although I haven't tested it since this is code that I just moved from one file to another so I'm not sure. 

For context, this panel here is used for Snyk Code within the integration with LS, which is a new feature that not many customers have seen.

We have also recently migrated this to HTML, so the fix should ideally be done there. We plan on testing this in this cycle for HTML so will leave it to then.